src/odemis/gui/cont/streams.py

```diff
@@ -2185,6 +2185,21 @@ def removeStream(self, stream):
         # would prevent the Stream render thread from terminating.
         gc.collect()
 
+    def clear_panel(self):
+        """
+        Clear the stream bar from its GUI controls, and remove each stream controllers stream from its view
+        Must be called in the main GUI thread
+        """
+        # clear the graphical part
+        self._stream_bar.clear()
+        # Clear the stream from the stream controller view
+        for sc in self.stream_controllers:
+            if hasattr(sc, "view") and hasattr(sc.view, "removeStream"):
```

Comment:
Why wouldn't the stream have a .view attribute? It's set at init (possibly with None, but seems to always be set).